src/evidently/EvidentlyManager.ts

```diff
@@ -0,0 +1,280 @@
+import { Dispatch } from '../dispatch/Dispatch';
+import { Config } from '../orchestration/Orchestration';
+import { EventCache } from '../event-cache/EventCache';
+import { NIL_UUID } from '../sessions/SessionManager';
+import {
+    BatchEvaluateFeatureRequest,
+    BatchEvaluateFeatureResult,
+    ContextType,
+    EvaluationResult,
+    EvaluationResults,
+    EvidentlyRequest
+} from './types';
+
+export const MAX_EVIDENTLY_FEATURES_PER_EVENT = 10;
+
+const REQUEST_CACHE_KEY = 'lastRequest';
+const EVALS_CACHE_KEY = 'cachedEvaluations';
+
+export class EvidentlyManager {
+    private eventCache: EventCache;
+    private dispatchManager: Dispatch;
+    private config: Config;
+    private isEnabled: boolean;
+
+    private lastRequest?: EvidentlyRequest;
+    private cachedEvaluations: EvaluationResults = {};
+    private lastPromise: Promise<EvaluationResults> = Promise.resolve({});
+    private cacheValidCheck = 0;
+
+    constructor(
+        config: Config,
+        eventCache: EventCache,
+        dispatchManager: Dispatch
+    ) {
+        this.eventCache = eventCache;
+        this.dispatchManager = dispatchManager;
+        this.config = config;
+        this.isEnabled = config.evidentlyConfig.project !== undefined;
+    }
+
+    public loadEvaluations(request: EvidentlyRequest) {
```

Comment:
Here's what existing products (LaunchDarkly, Statsig, Split, Optimizely, etc) do:

```
client.initialize({
	userId: string
	userAttributes?: JSON
})
featureVariation = client.evaluateFeature(feature: string)
```

I don't want to walk through any one-way doors that bring us further away from the long-term picture here. Mainly this includes the naming for the APIs and types, since we can't change these after they're released since that will break backwards compatibility.


1. Judging from this pattern, customers don't want to have to specify features in the initialization (`LoadEvaluations` command), since they're already doing it in the evaluation command. Therefore, it will need to be optional in the long-term, in which case no evaluations will be "loaded". Better naming would be "configureFeatures" or "initialize" - something that implies nothing will necessarily need to be evaluated/fetched.
2. Customers want to evaluate only a single feature, so I think we should start with just the EvaluateFeature API (a single feature), rather than a batch call for P0.
3. I think the `EvidentlyRequest` type needs to match the naming of the initialization command - this might get confused with some other Evidently Request type.
4. I think the `GetEvaluation` comand needs to be `EvaluateFeature` since it should have the same result and side effects as the API. 


Here's an example of what I'm talking about:
```
FeatureConfiguration{
   entityId: string
   evaluationContext?: EvaluationContext
}

cwr.configureFeatures(featureConfiguration) // initialization
cwr.evaluateFeature(feature: string) => EvaluateFeatureResult // evaluation
```

Reply:
After syncing offline the conclusion was:
1. Change `loadEvaluations` to `initializeFeatures` and change `features` to `preloadFeatures`
2. Change `getEvaluations` to `evaluateFeature`, change argument to only take a single feature instead of a list of features